### PR TITLE
NAS-110616 / 21.08 / Print "Press Enter to bring Console setup menu." after Linux console …

### PR DESCRIPTION
**Pull request by themylogin**
…gets spammed with kernel messages

### CODE REVIEW

File: src/freenas/etc/netcli [hunk 1]
```diff
@@ -1619,17 +1629,43 @@ def main_menu():
...
+                last_kernel_message = time.monotonic()
+
+
+def press_enter_to_bring_console_setup_menu():
+    global last_kernel_message
+    while True:
+        if last_kernel_message is not None and time.monotonic() - last_kernel_message >= 3:
+            last_kernel_message = None
+            if is_in_main_menu_prompt:
+                print("Press Enter to bring Console setup menu.")
```
Comment (yocalebo) [inline src/freenas/etc/netcli:1658]: Am I understanding correctly that this now gets printed to the console every 3 seconds?

File: src/freenas/etc/netcli [hunk 2]
```diff
@@ -1619,17 +1629,43 @@ def main_menu():
...
 
 
+last_kernel_message = None
+
+
+def read_kmsg():
+    global last_kernel_message, main_menu_initialized
+    with open("/dev/kmsg") as f:
+        while True:
+            k = f.readline()
```
Comment (yocalebo) [inline src/freenas/etc/netcli:1647]: why `readline()` if we don't do anything with it?

**Comment by william-gr** [comment]
Just realize we really need to do this in CLI. By 21.08 netcli will be gone and replaced with midcli.

File: src/freenas/etc/netcli (hunk 1 above)
Comment (themylogin) [reply src/freenas/etc/netcli:1658]: No, it only gets printed after we receive kernel messages (in 3 seconds after last message). Then `last_kernel_message` will be `None` until a new message is received

**Comment by rick-mesta** [comment]
@yocalebo: has Vlad addressed all issues to your satisfaction ?

**Comment by yocalebo** [comment]
> @yocalebo: has Vlad addressed all issues to your satisfaction ?

No, @william-gr chimed in and said this needs to be done in `midcli` and not via `netcli` since it's expected that `netcli` will be removed in 21.08 so this PR can more than likely be closed.

**Comment by rick-mesta** [comment]
oh... thx for the ⚡ fast reply, Caleb. @william-gr / @themylogin : you guys okay with closing this PR so the fix can be done in `midcli` ?

**Comment by william-gr** [comment]
yup

**Comment by rick-mesta** [comment]
As mentioned by both @william-gr and @yocalebo, this work should be delivered as part of `midcli`. Closing this PR.

**Pull request closed without merging**